common/src/main/cpp/extensions/openxr_fb_spatial_entity_sharing_extension_wrapper.cpp

```diff
@@ -0,0 +1,130 @@
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#include "extensions/openxr_fb_spatial_entity_sharing_extension_wrapper.h"
+
+#include <godot_cpp/classes/object.hpp>
+#include <godot_cpp/classes/open_xrapi_extension.hpp>
+#include <godot_cpp/templates/vector.hpp>
+#include <godot_cpp/variant/utility_functions.hpp>
+
+using namespace godot;
+
+OpenXRFbSpatialEntitySharingExtensionWrapper *OpenXRFbSpatialEntitySharingExtensionWrapper::singleton = nullptr;
+
+OpenXRFbSpatialEntitySharingExtensionWrapper *OpenXRFbSpatialEntitySharingExtensionWrapper::get_singleton() {
+	if (singleton == nullptr) {
+		singleton = memnew(OpenXRFbSpatialEntitySharingExtensionWrapper());
+	}
+	return singleton;
+}
+
+OpenXRFbSpatialEntitySharingExtensionWrapper::OpenXRFbSpatialEntitySharingExtensionWrapper() :
+		OpenXRExtensionWrapperExtension() {
+	ERR_FAIL_COND_MSG(singleton != nullptr, "An OpenXRFbSpatialEntitySharingExtensionWrapper singleton already exists.");
+
+	request_extensions[XR_FB_SPATIAL_ENTITY_SHARING_EXTENSION_NAME] = &fb_spatial_entity_sharing_ext;
+	singleton = this;
+}
+
+OpenXRFbSpatialEntitySharingExtensionWrapper::~OpenXRFbSpatialEntitySharingExtensionWrapper() {
+	cleanup();
+}
+
+void OpenXRFbSpatialEntitySharingExtensionWrapper::_bind_methods() {
+	ClassDB::bind_method(D_METHOD("is_spatial_entity_sharing_supported"), &OpenXRFbSpatialEntitySharingExtensionWrapper::is_spatial_entity_sharing_supported);
+}
+
+void OpenXRFbSpatialEntitySharingExtensionWrapper::cleanup() {
+	fb_spatial_entity_sharing_ext = false;
+}
+
+Dictionary OpenXRFbSpatialEntitySharingExtensionWrapper::_get_requested_extensions() {
+	Dictionary result;
+	for (auto ext : request_extensions) {
+		uint64_t value = reinterpret_cast<uint64_t>(ext.value);
+		result[ext.key] = (Variant)value;
+	}
+	return result;
+}
+
+void OpenXRFbSpatialEntitySharingExtensionWrapper::_on_instance_created(uint64_t instance) {
+	if (fb_spatial_entity_sharing_ext) {
+		bool result = initialize_fb_spatial_entity_sharing_extension((XrInstance)instance);
+		if (!result) {
+			UtilityFunctions::print("Failed to initialize fb_spatial_entity_sharing extension");
```

Comment:
Should be a `UtilityFunctions::printerr`

Reply:
All updated to `UtilityFunctions::printerr()` in my latest push!